UnitsNet/GeneratedCode/Quantities/Ratio.g.cs

```diff
@@ -822,28 +788,32 @@ private bool TryToUnit(RatioUnit unit, [NotNullWhen(true)] out Ratio? converted)
             return true;
         }
 
-        /// <inheritdoc />
-        IQuantity IQuantity.ToUnit(Enum unit)
-        {
-            if (!(unit is RatioUnit typedUnit))
-                throw new ArgumentException($"The given unit is of type {unit.GetType()}. Only {typeof(RatioUnit)} is supported.", nameof(unit));
-
-            return ToUnit(typedUnit, DefaultConversionFunctions);
-        }
-
         /// <inheritdoc cref="IQuantity.ToUnit(UnitSystem)"/>
         public Ratio ToUnit(UnitSystem unitSystem)
         {
             if (unitSystem is null)
                 throw new ArgumentNullException(nameof(unitSystem));
 
-            var unitInfos = Info.GetUnitInfosFor(unitSystem.BaseUnits);
+            return ToUnit(BaseUnit);
+        }
+
+        #region Explicit implementations
 
-            var firstUnitInfo = unitInfos.FirstOrDefault();
-            if (firstUnitInfo == null)
-                throw new ArgumentException("No units were found for the given UnitSystem.", nameof(unitSystem));
+        double IQuantity.As(Enum unit)
+        {
+            if (unit is not RatioUnit typedUnit)
+                throw new ArgumentException($"The given unit is of type {unit.GetType()}. Only {typeof(RatioUnit)} is supported.", nameof(unit));
 
-            return ToUnit(firstUnitInfo.Value);
+            return As(typedUnit);
+        }
+
+        /// <inheritdoc />
+        IQuantity IQuantity.ToUnit(Enum unit)
+        {
+            if (!(unit is RatioUnit typedUnit))
+                throw new ArgumentException($"The given unit is of type {unit.GetType()}. Only {typeof(RatioUnit)} is supported.", nameof(unit));
+
+            return ToUnit(typedUnit, DefaultConversionFunctions);
```

Comment:
Out of scope of PR, but the call to `ToUnit()` looks a bit weird to me.

- `DefaultConversionFunctions` is an `UnitConverter` instance, should maybe be named `UnitConverter`
- Should this really be `public`? Should we not encourage `UnitsNetSetup.Default.UnitConverter`, or the shorthand `UnitConverter.Default` instead?
- I then realize, each quantity creates their own instance o` UnitConverter`, in addition to `UnitsNetSetup.Default.UnitConverter`, which creates another instance based on all quantities in `UnitConverter.CreateDefault`. This means, modifying `Angle.DefaultConversionFunctions`  does not modify `UnitsNetSetup.Default.UnitConverte` and they become inconsistent, and vice versa 👎 

Adding to #1200

Reply:
I've actually dealt with this already- my upcoming release has this property marked as [Obsolete] and it now returns the `UnitConverter.Default` instance (which works for all conversions)... I've therefore completely removed, the need for `switching` and removed all of the reflection code..

Reply:
Ok very nice